### ISSUE TITLE
Configuration BullMQ

### ISSUE THREAD

**Issue by motawy** [taskforcesh/bullmq#331]
I'm looking into bullmq to use it to retry failed jobs.
I have it running as a separate process only to reprocess failed jobs from a separate queue (rabbitmq).

The only thing I'm concerned is the fact that it looses the lock on a job and fails with failedReason undefined. 

The job is processing is post to addresses that sometimes might be unreachable. I set up the bullmq to retry with exponential backoff with 16 attempts with 1000ms of delay. That should take around 30h to fully fail. 

My question is: 
Is it normal for the job to fail with ```Error: Missing lock for job 1 failed```
Am I doing something wrong?

**Comment by manast** [comment]
Missing lock should never happen in normal circumstances. The only reason this should happen is if the worker is very CPU intensive (or is working on a CPU that is saturated by other processes) so that the worker is not able to renew the lock on the job. If this is not the case I would like to know more about your case, and ideally if you could write some code that reproduces the issue.

**Comment by motawy** [comment]
@manast Thanks for the reply! Unfortunately it's hard to share code since it's quite a big project where I decided to integrate bullmq. 

From your reply is very likely that was the case. The use case is the following. 
The system is a webhook service. There are 4 other services that send jobs through rabbitmq to this one. The issue comes after validating the payload and trying to make a post request to the various urls. 

I wanted to use bullmq as a retry service as we all know that rabbitmq has some limitations to that. 

Initially I integrated bullmq into the validation, and post request which that could have been why it was also taking quite a bit to move from one job to the next. 

Now I just use it in the case of failure after the post. So jobs are only sent onto bullmq queue when failed by rabbitmq handler. 

I would be interested in your opinion about this use case.
